conjur_api/providers/authn_authentication_strategy.py

```diff
@@ -59,13 +59,17 @@ async def authenticate(self, connection_info, ssl_verification_data) -> tuple[st
 
         # If the credential provider already has a valid API token, return it
         if self._is_authenticated(creds):
-            return str(creds.api_token), cast(datetime, creds.api_token_expiration)
+            return str(creds.api_token), self._str_to_datetime(creds.api_token_expiration)
 
         await self._ensure_logged_in(connection_info, ssl_verification_data, creds)
         api_token = await self._send_authenticate_request(ssl_verification_data, connection_info, creds)
 
         return api_token, self._calculate_token_expiration(api_token)
 
+    @staticmethod
+    def _str_to_datetime(api_token_expiration: str):
```

Comment:
Why do we need to convert `creds.api_token_expiration` from string to datetime? Isn't it already in datetime format? If not, maybe we should change the where this variable gets its value, so it will be datetime to begin with?

Reply:
`creds` in `is_authenticated` is from type `CredentialData`, and it holds this field as string.

I agree that there may be better approach, I just made the smallest fixes so the CLI could work.

Reply:
I tried what you suggested but it causes trouble when trying to save the CredentialData since expiration is not a string.
We can make a conversion function in `CredentialData` that callers will have to use before save